### DISCUSSION TITLE
🚀 Path to Launch

### DISCUSSION THREAD

**Discussion by bautistaaa** [typehero/typehero#510] [Announcements]
- [ ] Ensure moderation tools are in place(admin.typehero.dev)
- [ ] Tracks - See discussions
   - need to reach out to people after some are done
- [ ]    Challenges
  - focus on beginner - med 
  - but also want some for the hardcore
- [x] 3. Ingestion path for content (we will do this post launch)
- [x] 4. Localization 
- [ ] 4. Profiles
- [x] 5. Finalize Editor UI
- [x] 6. SEO (#705)
- [x] 7. Security Audit (https://github.com/typehero/typehero/pull/864)
- [x] 8. Privacy Policy (#680)
- [x] 9. Terms of Service (#680)

**Comment by bautistaaa** [comment]
feedback welcome

**Comment by jhechtf** [comment]
I'm working on getting the admin.typehero.dev stuff up and running better, but with the new job I'm a bit slow.

For the challenges, I wonder why we should focus on the beginner / medium. The type-challenges repo already has all the stuff. Me personally I find that going to something "hard" or "extreme" and finding my current level of info lacking is one of the biggest drivers to me leveling up my skills. 

Could you expand a bit on what you mean by #3? 

RE: The Editor UI: @PickleNik pls

**Reply by PickleNik** [reply]
bro idk typescript, i need those beginner ones, pls spare me

**Reply by PickleNik** [reply]
and im down to help out with Editor and Profile UIs

**Reply by jhechtf** [reply]
I'm not saying we _don't_ include the beginner ones, I'm saying that we shouldn't exclude the hard / extreme ones initially. Challenges of all levels should be included on launch. With the "beginner friendly" tracks out there, I'm sure we'll funnel a lot of people that way, but if we don't make the site interesting for the more experienced people we'll likely be written off. We could maybe even make a "Devil 😈" track that's literally just a collection of super freaking hard challenges, you know, for the try hards out there.

**Reply by PickleNik** [reply]
oh mayb

**Reply by bautistaaa** [reply]
we will have a track for hardcore ones. the truth is there are more users in the beginner-medium space.

there isnt much from type challenge repo we want to use that actually shows you real world challenges that will help you in the real world. there are a few that we will take(and will give credit to) but most will be homegrown as this is what will truly set us apart as an education platform

**Reply by jhechtf** [reply]
I mean some challenges aren't "real world" until they are. I had to do a lot of FAFO to figure out a use case i had at work where I need to get only the key names of certain types passed into a generic. The abstract of the work would make an excellent at least hard level challenge, and up until that point (~3 or 4 years) of working with TS I'd never had to do something that complicated before.

Challenges are meant to be ... well challenging. Sure, there's going to be more people in the beginner space. That's fine -- but why throw away all of these challenges we're getting for free? It feels equivalent to me of Leetcode getting rid of any "hard" problem in their database because most of their users are junior or mid level engineers. will most people use the hard/extreme leetcode problems? probably not. But they're still _there_

**Reply by bautistaaa** [reply]
we will pick and choose from type challenges. the issue is most of them aren't very real world and are strictly in the type world. we want some actual function definitions etc. not to mention you need to be pretty well versed to do the beginner ones.

ive done most of the type challenges and they dont really translate to the real world so we're trying to differentiate ourselves there. if you want those types of challenges you can go to type challenges. 

we will certainly have hardcore challenges. no one is saying those wont be there.

**Reply by jhechtf** [reply]
I don't understand why we're trying to differentiate ourselves when type challenges itself was plenty popular. Perhaps its my math background but I don't see the point in throwing away good challenges just because they aren't "useful." Plenty of stuff in math was discovered in the like 1700/1800s and none of it was "useful" until we needed cryptography. Everything is useful to me, it's all a matter of when.

I've got some thoughts on the whole "focusing on the real world" but right now it's late and everytime i try to write it out it reads mean to me, so I'll try again tomorrow.

**Comment by Hacksore** [comment]
> Ingestion path for content

Why don't we postpone that for after launch? Having a git/github workflow is going to take quite a while to get that all setup but we can tell the community we are working on it.

I do think it's a good idea to have the challenges in github but adding them to this repo is going to cause a LOT of noise for maintainers.

**Reply by bautistaaa** [reply]
agreed manual is fine for now. i also dont anticipate much traffic in the area outside of those who want to contribute to the content already

**Comment by jhechtf** [comment]
for "Terms of Service" I feel like a very legalese (perhaps we could get Alice to write it?) "Don't be an asshole" is fine, but @bautistaaa what were you thinking of on this one? Same thing for the Privacy policy, a nice ol "we aren't going to sell your data" deal should do it.

**Reply by PickleNik** [reply]
https://termly.io/

**Reply by bautistaaa** [reply]
yeah we can literally just use a template and insert our name tbh

**Reply by jhechtf** [reply]
Sounds good to me
